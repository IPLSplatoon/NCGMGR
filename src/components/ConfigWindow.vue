<template>
    <ipl-space
        class="m-b-8"
        color="secondary"
    >
        <div class="bold m-b-6">
            Dependency status
        </div>
        <dependency-checker />
    </ipl-space>
    <ipl-space color="secondary">
        <ipl-small-toggle
            :model-value="configStore.userConfig.enableErrorLog"
            :disabled="errorLogToggleDisabled"
            @update:model-value="onErrorLogToggleChange"
        >
            Enable error log<br>
            <ipl-label>The error log is useful for diagnosing technical issues.</ipl-label>
        </ipl-small-toggle>
    </ipl-space>
</template>

<script lang="ts">
import { defineComponent, ref } from 'vue'
import { IplLabel, IplSmallToggle, IplSpace } from '@iplsplatoon/vue-components'
import { computed } from 'vue'
import { useConfigStore } from '@/store/configStore'
import DependencyChecker from '@/components/DependencyChecker.vue'

export default defineComponent({
    name: 'ConfigWindow',

    components: { DependencyChecker, IplLabel, IplSmallToggle, IplSpace },

    setup () {
        const configStore = useConfigStore()
        const errorLogToggleDisabled = ref(false)

        return {
            configStore,
            errorLogToggleDisabled,
            async onErrorLogToggleChange(newValue: boolean) {
                errorLogToggleDisabled.value = true
                try {
                    await configStore.patch({
                        enableErrorLog: newValue
                    })
                } finally {
                    errorLogToggleDisabled.value = false
                }
            }
        }
    }
})
</script>
