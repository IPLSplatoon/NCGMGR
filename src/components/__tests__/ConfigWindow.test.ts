import { createTestingPinia, TestingPinia } from '@pinia/testing'
import { config, mount } from '@vue/test-utils'
import ConfigWindow from '@/components/ConfigWindow.vue'

describe('ConfigWindow', () => {
    let pinia: TestingPinia

    beforeEach(() => {
        pinia = createTestingPinia()
        config.global.plugins = [pinia]
    })

    config.global.stubs = {
        IplSmallToggle: true,
        DependencyChecker: true
    }

    it('matches snapshot', () => {
        const wrapper = mount(ConfigWindow)

        expect(wrapper.html()).toMatchSnapshot()
    })
})
