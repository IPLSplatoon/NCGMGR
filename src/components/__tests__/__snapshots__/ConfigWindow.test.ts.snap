// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfigWindow matches snapshot 1`] = `
"<div data-v-d255ffee="" class="ipl-space color-secondary m-b-8">
  <div class="bold m-b-6"> Dependency status </div>
  <dependency-checker-stub></dependency-checker-stub>
</div>
<div data-v-d255ffee="" class="ipl-space color-secondary">
  <ipl-small-toggle-stub label="" modelvalue="false" disabled="false"></ipl-small-toggle-stub>
</div>"
`;
